crates/actors/src/mempool_service/inner.rs

```diff
@@ -772,6 +773,20 @@ impl Inner {
         &mut self,
         tx: &T,
     ) -> Result<(), TxIngressError> {
+        // compute expected id from the signature
+        let expected_id = {
+            let bytes: [u8; 32] = keccak256(tx.signature().as_bytes()).into();
+            H256::from(bytes)
+        };
+
+        // check id is valid
+        if expected_id != tx.id() {
+            let mut mempool = self.mempool_state.write().await;
+            mempool.recent_invalid_tx.put(tx.id(), ());
+            debug!("txid mismatch with signature");
+            return Err(TxIngressError::InvalidSignature);
+        }
```

Comment:
```suggestion
        // check id is valid
        if !tx.is_signature_valid() {
            let mut mempool = self.mempool_state.write().await;
            mempool.recent_invalid_tx.put(tx.id(), ());
            debug!("txid mismatch with signature");
            return Err(TxIngressError::InvalidSignature);
        }
```

There's already a method for this on the `IrysTransactionCommon` type - `is_signature_valid()`

Reply:
This was already in use just below my new code - and was not checking the id correctly. Interesting!